### ISSUE TITLE
GET requests only offer limited URI length

### ISSUE THREAD

**Issue by holtgrewe** [exomiser/Exomiser#329]
We have it an issue with the REST API. GET requests encode all parameters in the URI and the maximal URI length is relatively strictly limited by Spring. This leads to error HTTP responses.

As already mentioned in #326 switching to POST would probably help. Allowing for configuration of maximal POST size would allow then help further (maybe this last point is already possible by setting some java configuration or environment variable, I'd have to double-check in the Spring docs).

**Comment by holtgrewe** [comment]
@julesjacobsen can I haz POST requests? (🍔)

This would make the API much more useable for larger gene sets. I'd really appreciate it, in particular if it's only the "right" 5 lines to change when having your boss-level Spring skills! :)

**Comment by julesjacobsen** [comment]
Sorry, @holtgrewe I missed this. You can always just do a GET without specifying the `genes` parameter to save having to collect a panel. This will return all the gene results. Then you can use `limit` to the top 50 or whatever results.

Would this work or do you have to specify a very long list of gene identifiers?

**Comment by holtgrewe** [comment]
Thanks for your answer. I'd like to specify the list of gene specifiers.

**Comment by julesjacobsen** [comment]
You can now `POST` to exomiser/api/prioritise/

```
{
    "phenotypes": [
        "HP:0001156",
        "HP:0001363",
        "HP:0011304",
        "HP:0010055"
    ],
    "genes": [
        2263
    ],
    "prioritiser": "hiphive",
    "prioritiserParams": "human,mouse,fish,ppi",
    "limit": 20
}
```

There is a new class PrioritiserRequest you can use for this purpose.

**Comment by holtgrewe** [comment]
Excellent, thanks! You're the best. Can't wait to upgrade. When would the next release be?

**Comment by julesjacobsen** [comment]
Sorry, this took a while to be available as an official release - it's in [12.1.0](https://github.com/exomiser/Exomiser/releases/tag/12.1.0)
